main.tf

```diff
@@ -15,6 +15,8 @@ locals {
       can(regex(".*hs-crypto.*", var.kms_key_crn)) ? "hs-crypto" : null
     )
   ) : null
+
+  parameters_enabled = var.existing_kms_instance_guid != null && var.kms_key_crn != null ? true : false
```

Comment:
this logic should use `kms_encryption_enabled` variable

Reply:
I am unsure why we need to use `kms_encryption_enabled` as the `parameters_enabled` is used in the `ibm_resource_instance`. I just wanted to see if `existing_kms_instance_guid` and `kms_key_crn` are present, and based on that either add `parameters` or do not add.